include/zephyr/drivers/adc/current_sense_shunt.h

```diff
@@ -27,7 +27,7 @@ struct current_sense_shunt_dt_spec {
 #define CURRENT_SENSE_SHUNT_DT_SPEC_GET(node_id)                                                   \
 	{                                                                                          \
 		.port = ADC_DT_SPEC_GET(node_id),                                                  \
-		.shunt_micro_ohms = DT_PROP(node_id, shunt_resistor_micro_ohms),                   \
+		.shunt_milli_ohms = DT_PROP(node_id, shunt_resistor_micro_ohms) / 1000,            \
```

Comment:
what if the micro ohms specified in DT is less than 1000 ?

Reply:
The same thing that currently happens if the micro ohms is specified as 0 currently.
Unfortunately because `CURRENT_SENSE_SHUNT_DT_SPEC_GET` is an initialiser, it is not possible to add a `BUILD_ASSERT` into the macro.
The other option is to add a check for 0 in the conversion function.
Do you have an example of anything specifying a micro-ohms less than 1000? Because this PR would affect them regardless. The assumption in this PR is that it is not really possible to characterise a resistor to this level of accuracy.

Reply:
@ualbertagreen can you weigh in about this since you added this binding, did you have a reason for it being specified in micro-ohms

this PR in the current state is going to make it so that the DT is specified in micro-ohm but the code will round it to the nearest milliohm, which doesn't seem good to me to have this discrepancy

Reply:
The reason it is specified in micro-ohms is that it is duplicated directly from the Linux bindings.
They do a whole bunch of maths to try and avoid overflowing the same calculation, which makes sense given their much larger compute resources:
https://github.com/torvalds/linux/blob/aa0743a229366e8c1963f1b72a1c974a9d15f08f/drivers/iio/afe/iio-rescale.c#L322-L359

Reply:
Sorry for the late response, but exactly what Jordan said. It's only this way because it's directly copied from the Linux bindings.

If working in milli-ohms is better for Zephyr then feel free to change it.

Reply:
How about just mentioning in the bindings that these properties use microohms because they were imported from Linux but in calculations the numbers are rounded down to milliohms and therefore values < 1000 are not supported (and will lead to division by 0 at runtime)?

Reply:
to be clear, it's more than just values less than 1000 affected, but that any value will be rounded to the nearest 1000 / nearest milliohm